docs/rfc/0002-webrc-client-api.md

```diff
@@ -0,0 +1,1612 @@
+7. Passing errors, [RTCPeerConnection]s, [RTCStatsReport]s.
+8. Cover both [p2p full mesh] and [hub server (SFU, MCU)] scenarios.
+
+The protocol must be versatile enough to cover all possible use cases.
+
+
+
+
+## Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+### What is `Client WebRTC API`? 
+
+It is a part of `Client API` responsible for [WebRTC] connection management. You can find `Client API` on the following approximate architecture design:
+```
+                                                                       .------------Server-----------.
+                                                                       :     .-------------------.   :
+                          .--------------------------------------------+-----o  Control Service  :   :
+                          :                                            :     '--------o----------'   :
+                          :                                            :              |              :
+                          :                                            :        Control Api          :
+.--------Client-----------+------------------------.                   :              |              :
+:  .--------------------. :  .--------------------. :  .-Client-API--. :  .-----------o------------. :
+:  :  User Application  o-'  :     Web Client     o-+--'             '-+--o      Media Server      : :
+:  :                    :----:                    o-+--.             .-+--o                        : :
+:  '--------------------'    '--------------------' :  '----Media----' :  '------------------------' :
+'---------------------------------------------------'                  '-----------------------------'
+```
+
+So, how it works from `Media Server` point of view:
+1. `Control Service` configures media room via `Control API`.  
+2. `Media Server` provides all necessary information (URLs + credentials) for all room members.
+3. `User Application` passes credentials and other necessary stuff (like `<video>` elements) to `Web Client`.
+4. And voilà!
+
+
+### Transport considerations
+
+Although, signalling can be implemented on top of any transport, WebSocket suits the most since it provides small overhead reliable duplex connection, is widely adopted and supported.
+
+
+### Protocol considerations
+
+Existing best practices are recommended for final implementation:
+1. Message level `ping`/`pong`, since it is the most reliable way to detect connection loss (protocol level WebSocket `ping`/`pong` may disfunct due to browser implementation and is not exposed to `Web Client`).
+2. Reconnects, since [RTCPeerConnection] always outlives WebSocket connection in cases of network issues, and both parts should know when to dispose related resources.
+3. Transactions, so each part will receive answers to all request sent, and be able to know if request was processed correctly or caused some kind of error.
```

Comment:
@alexlapa why this is called _Transaction_? Where this originates from? Generally, it's just request/response, so is kind of RPC.